### ISSUE TITLE
[question] DML - binary and continuous treatment

### ISSUE THREAD

**Issue by ieriii** [microsoft/EconML#597]
Thank you very much for your great library.
I would like to run DML (and linearDML) using multiple treatments: one of them is binary (1 and zero) the other ones are continuous.

Is there a way I can do this with your library? I checkde the source code but I cannot confirm whether this would be a possiblity.

Thank you.

**Comment by kbattocchi** [comment]
We do not directly support combinations of discrete and continuous features (or more than one discrete treatment at a time), so typically you would need to one-hot-encode the discrete treatments and drop one of them and then treat them as continuous.  In the case where the discrete treatment is binary, this is just equivalent to using the column directly, though, so for your case you should be able to just concatenate that column with your continuous treatment and pass it as a two-dimensional treatment array.

**Comment by ieriii** [comment]
Thank you for your prompt reply. 
I'm just concerned that with you suggestion I would end up using a regressor to predict a discrete variable. I'm not sure the impact that could have on the final estimates. let me know if I'm missing something.

Thank you

**Comment by kbattocchi** [comment]
Yes, you do need to use a single regressor that will be applied to all treatments; however, that could be a custom model that combines a classifier and a regressor under the hood, something like:

    import numpy as np

    class HybridModel:
        def __init__(self, classifier, regressor):
            self.classifier = classifier
            self.regressor = regressor

        def fit(self, X, Y):
            assert Y.ndim == 2 and Y.shape[1] == 2
            self.classifier.fit(X, Y[:,0])  # assume categorical column comes first
            self.regressor.fit(X, Y[:,1:])
            return self

        def predict(self, X):
            return np.hstack([self.classifier.predict_proba(X)[:,1:], self.regressor.predict(X)])

    from sklearn.linear_model import LogisticRegression, LinearRegression

    X = np.random.rand(100, 5)
    t1 = np.random.choice([0, 1], size=100)
    t2 = np.random.rand(100)
    Y = np.stack([t1, t2], axis=1)

    HybridModel(LogisticRegression(), LinearRegression())

with the caveat that I've only minimally tested this.

**Comment by ieriii** [comment]
Oh right. This is very interesting. 
Thank you for the hint. I'll give it a go and see how it goes. I'll close the issue for now since it is not a proper issue.

And thank you very much. Very kind!
